org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/GlobalTrustManager.java

```diff
@@ -77,16 +84,44 @@ public void installGlobalTrustManager(String protocol, String tmAlgorithm) {
     X509TrustManager globalTrustManager;
     try {
       globalTrustManager = createGlobalTrustManager(tmAlgorithm, getAllTrustedCertificates());
-
       SSLContext sslContext = SSLContext.getInstance(protocol);
-      sslContext.init(null, new TrustManager[]{globalTrustManager}, SecurityUtility.createSecureRandom());
+      sslContext.init(loadKeyManagers(), new TrustManager[]{globalTrustManager}, SecurityUtility.createSecureRandom());
       SSLContext.setDefault(sslContext);
     }
     catch (Exception e) {
       throw new ProcessingException("could not install global trust manager.", e);
     }
   }
 
+  /**
+   * Loads a list of {@link KeyManager} based on {@code javax.net.ssl.*} system properties.
+   *
+   * @see sun.security.ssl.SSLContextImpl.DefaultManagersHolder#getKeyManagers()
+   */
+  @SuppressWarnings("squid:S1168")
+  protected KeyManager[] loadKeyManagers() throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException, UnrecoverableKeyException, NoSuchProviderException {
+    String keystoreLocation = System.getProperty("javax.net.ssl.keyStore");
+    if (StringUtility.hasText(keystoreLocation)) {
+      String keyStoreType = System.getProperty("javax.net.ssl.keyStoreType");
+      String keyStoreProvider = System.getProperty("javax.net.ssl.keyStoreProvider");
+      KeyStore keyStore;
+      if (StringUtility.hasText(keyStoreProvider)) {
+        keyStore = KeyStore.getInstance(keyStoreType, keyStoreProvider);
+      }
+      else {
+        keyStore = KeyStore.getInstance(keyStoreType);
+      }
+      char[] keystorePassword = System.getProperty("javax.net.ssl.keyStorePassword", "").toCharArray();
+      try (FileInputStream keystoreInputStream = new FileInputStream(keystoreLocation)) {
+        keyStore.load(keystoreInputStream, keystorePassword);
+      }
+      KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+      keyManagerFactory.init(keyStore, keystorePassword);
+      return keyManagerFactory.getKeyManagers();
+    }
+    return null;
+  }
+
```

Comment:
How about supporting keyStoreType "PKCS11" (token)? It requires "javax.net.ssl.keyStore" to have the value "NONE" and must not provide a password.